api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -425,13 +429,23 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
   @Override
   public ResponseEntity<List<FileDetail>> getNoteBookList(String workspaceNamespace,
       String workspaceId) {
-    List<FileDetail> fileList = new ArrayList<>();
+    final List<FileDetail> fileList;
     try {
       org.pmiops.workbench.firecloud.model.Workspace fireCloudWorkspace =
           fireCloudService.getWorkspace(workspaceNamespace, workspaceId)
               .getWorkspace();
       String bucketName = fireCloudWorkspace.getBucketName();
       fileList = getFilesFromNotebooks(bucketName);
+      if (fileList != null && fileList.size() > 0) {
+        long userId = userProvider.get().getUserId();
+        CompletableFuture.runAsync(() -> {
```

Comment:
This is neat - is this the first time we're doing this in the codebase? I have a few questions about what's happening here:

- What thread is this running on?
- Does this run after the request returns?
- Are we guaranteed that the UserProvider will still have a reference to that previously authenticated user?
- How does this interact with unit tests? Does this get executed before the unit test completes?

+@danqrodney in case you have any Spring knowledge on the subject.

Reply:
My understanding of `runAsync` is that it does not spin off a new or different thread, so would be running in the current thread.

Reply:
Other questions are ones I have as well.

Reply:
OK, I'm not entirely convinced that the request-related information is guaranteed to be properly bound in this callback, but at least it seems your callback currently doesn't require any.

I would feel a bit better if you could figure out how to test this though. One possible approach:
- Inject an Executor into this controller to pass to `runAsync`.
- In real code, bind that to `ThreadPoolTaskExecutor`, which appears to be the default recommendation from Spring: https://docs.spring.io/spring/docs/current/spring-framework-reference/integration.html#scheduling
- In tests, bind that to `SyncTaskExecutor`

See also this thread for how to setup the binding in the real code: https://dzone.com/articles/spring-and-threads-taskexecutor

Alternatively, if you don't feel like figuring this out right now you could just drop the runAsync for now (though I do like the idea here).

Reply:
I did add the test surrounding this functionality, but after looking at @UrsaStutsman PR for notebook actions, i feel handling orphan notebooks are not going to add much (since delete and other functionality are being handled in 857 and we have another card 1195 to integrate these functions with userRecentResource). Thoughts?